### PR TITLE
MGDAPI-3855  - Implement persistent volume for alert manager

### PR DESCRIPTION
**Pull request by valerymo**
# Issue link
Implement persistent volume for alert manager
https://issues.redhat.com/browse/MGDAPI-3855
Replace PR https://github.com/integr8ly/integreatly-operator/pull/2646 (because of multiple rebases git issue)
This PR could be merged as Observability PR (that part of this Jira): https://github.com/redhat-developer/observability-operator/pull/114 is already merged and New OO release available: v3.0.11.

# What
- Implement persistent volume for alert manager
- This PR includes change of OO version: from v3.0.8 to v3.0.11.

# Verification steps
1. Check that Alertmanager PVC created in redhat-rhoam-observability namespace. For example:
$ oc get pvc -n redhat-rhoam-observability
NAME STATUS VOLUME CAPACITY ACCESS MODES STORAGECLASS AGE
alertmanager-alertmanager-db-alertmanager-alertmanager-0 Bound pvc-60903afd-b8b0-43c1-8fc4-1adc4062fe01 1Mi RWO gp2 3h2m
prometheus-prometheus-db-prometheus-prometheus-0 Bound pvc-4ae289f2-5fa8-42f1-9864-19aa805e0760 50Gi RWO gp2 3h2m

2. Enter Alertmanager UI, Silent/Snooze any alert (for example alertname="DeadMansSwitch"), kill alertmanager pod , wait for recreation of the pod, enter again Alertmanager UI, and see that alert still snoozed

3. Notes: - The PR was verified on local installation and install from bundle  - https://quay.io/repository/vmogilev_rhmi/managed-api-service-index?tab=tags&tag=1.23.0

### CODE REVIEW

**Comment by valerymo** [comment]
Verification steps done, on server `valerym02`
@MStokluska following our discussions, could you please approve this PR

CC @makslion @KevFan  - PR was recreated , Old PR https://github.com/integr8ly/integreatly-operator/pull/2646 with review notes (all addressed) was closed, as old PR was corrupted.

File: manifests/integreatly-observability-operator/3.0.11/observability-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+  links:
+  - name: Observability Operator
+    url: https://observability-operator.domain
+  maintainers:
+  - email: mas-users@redhat.com
+    name: rhosak
+  maturity: alpha
+  provider:
+    name: Red Hat
+  replaces: observability-operator.v3.0.8
```
Comment (MStokluska) [inline manifests/integreatly-observability-operator/3.0.11/observability-operator.clusterserviceversion.yaml:376]: can we please change it to 3.0.10

**Comment by MStokluska** [comment]
hey @valerymo just noticed, if we are bumping OO we should also bump https://github.com/integr8ly/integreatly-operator/blob/master/apis/v1alpha1/rhmi_types.go#L106
And tests: https://github.com/integr8ly/integreatly-operator/blob/master/test/common/operator_versions.go#L34

I would also investigate/create a follow up and try to understand why did the e2e passed with wrong OO version?

**Comment by valerymo** [comment]
@MStokluska , @makslion I'm sorry for so many attempts to proceed with this PR, and for small but very important things, that need to be added. Thank you very much for your Comments and useful notes and hope that now it's more close to be approved : ). cc @KevFan

**Pull request merged**